iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinServiceClientTests.java

```diff
@@ -1,38 +1,126 @@
+import com.microsoft.azure.sdk.iot.service.digitaltwin.DigitalTwinClient;
+import com.microsoft.azure.sdk.iot.service.digitaltwin.models.BasicDigitalTwin;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.*;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import tests.integration.com.microsoft.azure.sdk.iot.digitaltwin.helpers.E2ETestConstants;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.IntegrationTest;
+import tests.integration.com.microsoft.azure.sdk.iot.helpers.Tools;
 import tests.integration.com.microsoft.azure.sdk.iot.helpers.annotations.DigitalTwinTest;
+import tests.integration.com.microsoft.azure.sdk.iot.iothub.twin.TwinPnPTests;
 
 import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.UUID;
+
+import static com.microsoft.azure.sdk.iot.device.IotHubClientProtocol.MQTT;
+import static com.microsoft.azure.sdk.iot.device.IotHubClientProtocol.MQTT_WS;
+import static java.util.Arrays.asList;
+import static org.junit.Assert.assertEquals;
 
 @DigitalTwinTest
-public class DigitalTwinServiceClientTests {
+@Slf4j
+@RunWith(Parameterized.class)
+public class DigitalTwinServiceClientTests extends IntegrationTest
+{
+
+    private static final String IOTHUB_CONNECTION_STRING = Tools.retrieveEnvironmentVariableValue(E2ETestConstants.IOTHUB_CONNECTION_STRING_ENV_VAR_NAME);
+    private static RegistryManager registryManager;
+    private String deviceId;
+    private DeviceClient deviceClient;
+    private DigitalTwinClient digitalTwinClient = null;
+    private static final String DEVICE_ID_PREFIX = "DigitalTwinServiceClientTests_";
+    public TwinPnPTests.TwinPnPTestInstance testInstance;
+
+    @Rule
+    public Timeout globalTimeout = Timeout.seconds(5 * 60); // 5 minutes max per method tested
+
+    @Parameterized.Parameter(0)
+    public IotHubClientProtocol protocol;
+
+    @Parameterized.Parameters(name = "{index}: Digital Twin Test: protocol={0}")
+    public static Collection<Object[]> data() {
+        return asList(new Object[][] {
+                {MQTT},
+                {MQTT_WS},
+        });
+    }
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws IOException {
+        registryManager = RegistryManager.createFromConnectionString(IOTHUB_CONNECTION_STRING);
+    }
+
+    @Before
+    public void setUp() throws URISyntaxException, IOException, IotHubException {
+
+        this.deviceClient = createDeviceClient(protocol);
+        deviceClient.open();
+        DigitalTwinAsyncClient asyncClient = new DigitalTwinAsyncClient(IOTHUB_CONNECTION_STRING);
+        digitalTwinClient = new DigitalTwinClient(asyncClient);
+    }
+
+    @After
+    public void cleanUp() {
+        try {
+            deviceClient.closeNow();
+            registryManager.removeDevice(deviceId);
+            registryManager.close();
```

Comment:
Remove registryManager.close(); - This should only be done in @AfterClass - line 99

Reply:
Totally agree. Fixed it